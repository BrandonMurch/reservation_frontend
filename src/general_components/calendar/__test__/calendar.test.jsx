// Dependencies
import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import { render, fireEvent } from '@testing-library/react';
import { create } from 'react-test-renderer';
import moment from 'moment';

// Components
import Calendar from '../index';

const getMonth = function getMonthName(monthIndex) {
  return moment.months()[monthIndex];
};

describe('<Calendar />', () => {
  let component;
  let container = null;
  const mockOnDateRender = jest.fn();

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    component = render(<Calendar onDateRender={mockOnDateRender} />);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('should match snapshot', () => {
    const tree = create(<Calendar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onDateRender', () => {
    expect(mockOnDateRender).toHaveBeenCalled();
  });

  it('should go to the next month when the button is pressed', () => {
    const dateObject = moment();
    const button = component.getByRole('button', { name: '>' });
    fireEvent.click(button);
    // 2 months is due to React.StrictMode calling the reducer twice in development.
    dateObject.add(2, 'months');
    const dateText = component.getByText(new RegExp(getMonth(dateObject.get('months'))));
    expect(dateText).toBeInTheDocument();
  });

  it('should go to the previous month when the button is pressed', () => {
    const dateObject = moment();
    const button = component.getByRole('button', { name: '<' });
    fireEvent.click(button);
    // 2 months is due to React.StrictMode calling the reducer twice in development.
    dateObject.subtract(2, 'months');
    const dateText = component.getByText(new RegExp(getMonth(dateObject.get('months'))));
    expect(dateText).toBeInTheDocument();
  });

  it('should go to the current month when the button is pressed', () => {
    const dateObject = moment();
    let button = component.getByRole('button', { name: '>' });
    fireEvent.click(button);
    fireEvent.click(button);
    button = component.getByRole('button', { name: 'Today' });
    fireEvent.click(button);
    const dateText = component.getByText(new RegExp(getMonth(dateObject.get('months'))));
    expect(dateText).toBeInTheDocument();
  });

  it('should render titles for days of week', () => {
    const monday = component.getByText(/Mon/i);
    expect(monday).toBeInTheDocument();
    const friday = component.getByText(/Fri/i);
    expect(friday).toBeInTheDocument();
  });
});
