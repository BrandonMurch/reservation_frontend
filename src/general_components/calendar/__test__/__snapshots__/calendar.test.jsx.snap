// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <div>
      <h1
        className="monthText"
      >
        August 2020
      </h1>
    </div>
    <div
      className="buttonContainer"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        &lt;
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        &gt;
      </button>
    </div>
  </div>
  <table
    className="table"
    role="grid"
  >
    <thead>
      <tr>
        <th
          className="column"
        >
          Sun
        </th>
        <th
          className="column"
        >
          Mon
        </th>
        <th
          className="column"
        >
          Tue
        </th>
        <th
          className="column"
        >
          Wed
        </th>
        <th
          className="column"
        >
          Thu
        </th>
        <th
          className="column"
        >
          Fri
        </th>
        <th
          className="column"
        >
          Sat
        </th>
      </tr>
    </thead>
    <tbody
      className="calendarBody"
    >
      <tr />
      <tr>
        <td
          className="container"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="container"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="container"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="container"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="container"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="container"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            1
          </p>
          <p
            className="messageText"
          />
        </td>
      </tr>
      <tr>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            2
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            3
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            4
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            5
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            6
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            7
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            8
          </p>
          <p
            className="messageText"
          />
        </td>
      </tr>
      <tr>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            9
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            10
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            11
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            12
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            13
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="todayContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            14
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            15
          </p>
          <p
            className="messageText"
          />
        </td>
      </tr>
      <tr>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            16
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            17
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            18
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            19
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            20
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            21
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            22
          </p>
          <p
            className="messageText"
          />
        </td>
      </tr>
      <tr>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            23
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            24
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            25
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            26
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            27
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            28
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            29
          </p>
          <p
            className="messageText"
          />
        </td>
      </tr>
      <tr>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            30
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="activeContainer"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            31
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="container"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="container"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="container"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="container"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            
          </p>
          <p
            className="messageText"
          />
        </td>
        <td
          className="container"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="gridcell"
        >
          <p
            className="dateText"
          >
            
          </p>
          <p
            className="messageText"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
